src/components/Prescriptions.tsx

```diff
@@ -18,6 +21,56 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
   onEdit
 }) => {
   const { t } = useTranslationHook()
+  const { token } = useContext(TokenContext)
+
+  const [filteredPrescriptions, setFilteredPrescriptions] = useState<Prescription[]>(prescriptions)
+
+  // Using Record to get a key-value array
+  const [patients, setPatients] = useState<Record<number, Patient | null>>({})
+
+  useEffect(() => {
+    const fetchPatients = async (): Promise<void> => {
+      assert(token)
+      try {
+        // Create a promise for each prescription
+        const patientPromises = prescriptions.map(async (prescription) => {
+          const patientData = await getPatient(token, prescription.patient)
+
+          // Returns an object with patient data and prescription id
+          return { id: prescription.id, patient: patientData }
+        })
+        // Waits for patientPromises to be loaded
+        const patientResults = await Promise.all(patientPromises)
+
+        const patientsData: Record<number, Patient | null> = {}
+
+        patientResults.forEach(({ id, patient }) => {
+          patientsData[id] = patient
+        })
+
+        setPatients(patientsData)
+      } catch (error) {
+        console.error(error)
+      }
+    }
```

Comment:
So I imagine this is in preparation for "prescription full text search"

Reply:
exactly